src/ng/directive/ngModel.js

```diff
@@ -1095,141 +1251,97 @@ var DEFAULT_REGEXP = /(\s+|^)default(\s+|$)/;
  *   - `allowInvalid`: boolean value which indicates that the model can be set with values that did
  *     not validate correctly instead of the default behavior of setting the model to undefined.
  *   - `getterSetter`: boolean value which determines whether or not to treat functions bound to
-       `ngModel` as getters/setters.
+ *     `ngModel` as getters/setters.
  *   - `timezone`: Defines the timezone to be used to read/write the `Date` instance in the model for
  *     `<input type="date">`, `<input type="time">`, ... . Right now, the only supported value is `'UTC'`,
  *     otherwise the default timezone of the browser will be used.
+ */
+var ngModelOptionsDirective = ['$modelOptions', function($modelOptions) {
+  return {
+    restrict: 'A',
+    // ngModelOptions needs to run before ngModel and input directives
+    priority: 10,
+    require: ['ngModelOptions', '?^^ngModelOptions'],
```

Comment:
This means every `ngModelOptions` directive needs to search for a parent `ngModelOptions` up to the app root element. Isn't that a bit excessive? I think we could instead allow `ngModelOptions` on form / ngForm directives, which also seems more semantic to me.